### ISSUE TITLE
Broken node packages due to bad tarballs

### ISSUE THREAD

**Issue by pacien** [NixOS/nixpkgs#62632]
## Issue description

Some node packages are broken due to erroneous permissions in some tarballs, such as missing directory traversal/execution bits on directories.

### Steps to reproduce

Attempt building one of the affected package, such as `nodePackages.tedicross`, which depends on the broken `discord.js` tarball (https://github.com/discordjs/discord.js/issues/3322).

```
❯ nix-build '<nixos-unstable>' -A nodePackages.tedicross                                                                                                                    
these derivations will be built:                                                                                                                                            
  /nix/store/3sf4w9s3869ia8wsxsvxr0vkl5fazvvr-node-tedicross-0.8.7.drv                                                                                                      
building '/nix/store/3sf4w9s3869ia8wsxsvxr0vkl5fazvvr-node-tedicross-0.8.7.drv'...                                                                                          
unpacking sources                                                                                                                                                           
patching sources                                                                                                                                                            
configuring                                                                                                                                                                 
no configure script, doing nothing                                                                                                                                          
building                                                                                                                                                                    
installing                                                                                                                                                                  
[... other tarballs being unpacked successfully ...]                                                                              
unpacking source archive /nix/store/8bi171mspwk9iv9jy9jmhm1qhgalxbwr-discord.js-11.5.0.tgz                                                                                  
/nix/store/qp8rwgjnxzj5gn56ypqkl4xvdah9wzcz-gnutar-1.32/bin/tar: package/src/client: Cannot mkdir: Permission denied
[... many permission errors due to invalid permissions in the tarball ...]
/nix/store/qp8rwgjnxzj5gn56ypqkl4xvdah9wzcz-gnutar-1.32/bin/tar: package/src/client/websocket/packets/handlers/WebhooksUpdate.js: Cannot open: Permission denied
/nix/store/qp8rwgjnxzj5gn56ypqkl4xvdah9wzcz-gnutar-1.32/bin/tar: Exiting with failure status due to previous errors
do not know how to unpack source archive /nix/store/8bi171mspwk9iv9jy9jmhm1qhgalxbwr-discord.js-11.5.0.tgz
builder for '/nix/store/3sf4w9s3869ia8wsxsvxr0vkl5fazvvr-node-tedicross-0.8.7.drv' failed with exit code 1
error: build of '/nix/store/3sf4w9s3869ia8wsxsvxr0vkl5fazvvr-node-tedicross-0.8.7.drv' failed
```

## Technical details

```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.45, NixOS, 19.03.172732.f70e66cccd5 (Koi)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.03.172786.606306e0eaa, nixos-unstable-19.09pre181629.ae71c13a92f"`
 - channels(kea): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by pacien** [comment]
A patch mitigating this issue has been merged in `node2nix` (https://github.com/svanderburg/node2nix/pull/126 and https://github.com/svanderburg/node2nix/commit/4054c148edb6a1926eeab987803fe4777101c8a1).

Repairing the broken packages in `unstable` might require updating `node2nix` in nixpkgs. CC @svanderburg.

**Comment by pacien** [comment]
Solved by 18c2b52.
